### ISSUE TITLE
Level seems to have no effect

### ISSUE THREAD

**Issue by asbjornu** [reviewdog/action-shellcheck#9]
I've configured `level: style` and while that produces quite a few errors locally, `reviewdog/action-shellcheck` [does not fail the build](https://github.com/SwedbankPay/git-directory-deploy/pull/1/checks?check_run_id=685370494#step:4:1). None of the errors I see locally are printed anywhere by `reviewdog/action-shellcheck`. Have I configured it wrong, somehow?

```yaml
- name: Shellcheck
  uses: reviewdog/action-shellcheck@v1
  with:
    github_token: ${{ secrets.GITHUB_TOKEN }}
    reporter: github-pr-review
    level: style
```

**Comment by haya14busa** [comment]
level is the same as reviewdog's `-level` flag and style is not valid option. https://github.com/reviewdog/action-shellcheck#level

**Comment by haya14busa** [comment]
You can use `shellcheck_flags` instead.

**Comment by asbjornu** [comment]
I see. Thanks. I tried setting `level` to `info` and adding `shellcheck_flags: --severity=style` but it still doesn't detect any faults in my `.sh` scripts.

```yaml
- name: Shellcheck
    uses: reviewdog/action-shellcheck@v1
    with:
      github_token: ${{ secrets.GITHUB_TOKEN }}
      reporter: github-pr-review
      level: info
      shellcheck_flags: --severity=style
```

I'm also a bit unsure how `shellcheck_flags` works. It says that its default value is `--external-sources`, so if I set `shellcheck_flags` to something else, does that mean `--external-sources` is unset? If that's the case, the check should fail for doing `source` on external sources as well, which it doesn't.

**Comment by LaurenceGA** [comment]
> I'm also a bit unsure how shellcheck_flags works. It says that its default value is --external-sources, so if I set shellcheck_flags to something else, does that mean --external-sources is unset?

Yeah I think that's correct. You [can see](https://github.com/reviewdog/action-shellcheck/blob/master/entrypoint.sh#L9) they're passed as `${INPUT_SHELLCHECK_FLAGS:-'--external-sources'}`, so I think a custom value will override it. Not sure about why you're not seeing your desired outcome.

[This](https://github.com/SwedbankPay/git-directory-deploy/pull/4) is your current PR to test?
[Under the check](https://github.com/SwedbankPay/git-directory-deploy/pull/4/checks?check_run_id=689053421) it says:
> reviewdog: This is Pull-Request from forked repository.
GitHub token doesn't have write permission of Review API, so reviewdog will
report results via logging command [1] and create annotations similar to
github-pr-check reporter as a fallback.

You can see it in the logs
That explains why you aren't seeing a PR review.
It seems like it has [correctly annotated this line](https://github.com/SwedbankPay/git-directory-deploy/pull/4/files#diff-60254338249f657a0a83f98258a56bfeR216) that I assume you changed for testing purposes.

Were you expecting the check to fail? 🤔 It wouldn't if you had the level set to `warning` or `info`, only `error` would fail.

**Comment by asbjornu** [comment]
>You [can see](https://github.com/reviewdog/action-shellcheck/blob/master/entrypoint.sh#L9) they're passed as `${INPUT_SHELLCHECK_FLAGS:-'--external-sources'}`, so I think a custom value will override it. Not sure about why you're not seeing your desired outcome.

If `--external-sources` isn't passed to `shellcheck`, shouldn't the entire check fail with a big error when it discovers a `source` command in the `.sh` file?

> [This](https://github.com/SwedbankPay/git-directory-deploy/pull/4) is your current PR to test?

Correct.

> [Under the check](https://github.com/SwedbankPay/git-directory-deploy/pull/4/checks?check_run_id=689053421) it says:
> 
> > reviewdog: This is Pull-Request from forked repository.
> > GitHub token doesn't have write permission of Review API, so reviewdog will
> > report results via logging command [1] and create annotations similar to
> > github-pr-check reporter as a fallback.

Thanks for pointing this out. It is wrong, though. The PR is not from a forked repository, but from a [local branch](https://github.com/SwedbankPay/git-directory-deploy/tree/feature/shellcheck-severity).

> It seems like it has [correctly annotated this line](https://github.com/SwedbankPay/git-directory-deploy/pull/4/files#diff-60254338249f657a0a83f98258a56bfeR216) that I assume you changed for testing purposes.

Thanks for pointing out that as well, I expected to see the pull request comments show up in the "Conversation" tab of the PR. It's great that this works, though! Is it possible to have the annotations also be visible on the "Conversation" tab like ordinary code review comments?

> Were you expecting the check to fail? 🤔 It wouldn't if you had the level set to `warning` or `info`, only `error` would fail.

I would expect the GitHub Action to reflect the behaviour of the `shellcheck` CLI. That is, if I replace `uses: reviewdog/action-shellcheck` with `run: shellcheck *.sh`, I should see the exact same behaviour from both.

When I execute `shellcheck --severity=error *.sh` it will return `0` (i.e. succeed) unless there are errors in my shell scripts. If I notch the `severity` down to `warning`, `shellcheck` will start returning `1` (i.e. fail) for all warnings it finds. The same for `info` and eventually `style`, which will make `shellcheck` fail for every itsy little minute detail that may be improved in my scripts, including everything up through `info`, `warning` and `error`.

**Comment by haya14busa** [comment]
@asbjornu 
Use `fail_on_error`. https://github.com/reviewdog/action-shellcheck#fail_on_error

> Thanks for pointing this out. It is wrong, though. The PR is not from a forked repository, but from a local branch.

Interesting. Thanks for the report. It's actually a bug https://github.com/reviewdog/reviewdog/issues/603.

**Comment by asbjornu** [comment]
> Use `fail_on_error`. https://github.com/reviewdog/action-shellcheck#fail_on_error

Thanks. I've now configured `fail_on_error: true`, but since I'm unable to adjust ShellCheck's `severity`, [the build still doesn't fail](https://github.com/SwedbankPay/git-directory-deploy/actions/runs/114478122). It seems like [`shellcheck_flags` isn't a valid input](https://github.com/SwedbankPay/git-directory-deploy/actions/runs/109288485):

```
Unexpected input 'shellcheck_flags', valid inputs are ['github_token', 'level', 'reporter', 'path', 'pattern', 'exclude']
```

<img width="1072" alt="image" src="https://user-images.githubusercontent.com/12283/82787125-12f32980-9e66-11ea-8e44-2f4bcf3b49d0.png">

Ideas?

> Interesting. Thanks for the report. It's actually a bug [reviewdog/reviewdog#603](https://github.com/reviewdog/reviewdog/issues/603).

I see, that makes sense given that `isPRFromForkedRepo` just returns the value of the `event.PullRequest.Head.Repo.Fork` property. There should probably be added a comparison between the PR's source and target repositories as well.

**Comment by haya14busa** [comment]
Fixed shell check flags issue. #11

Can you try again?

**Comment by asbjornu** [comment]
Ok, great! But I don't think I can effectively test this until reviewdog/reviewdog#603 is fixed?

**Comment by haya14busa** [comment]
Fixed. #12

**Comment by asbjornu** [comment]
[It now seems to be working fine](https://github.com/SwedbankPay/git-directory-deploy/pull/4#pullrequestreview-449939593):

<img width="922" alt="Reviewdog PR comment" src="https://user-images.githubusercontent.com/12283/87687952-a40bc000-c786-11ea-8b4b-5ef11b7ae4e7.png">


Thanks, @haya14busa! 🙏
